File: plugins/discourse-adplugin/assets/javascripts/discourse/models/ad-type.js
import { computed } from "@ember/object";

export default class AdType {
  constructor(id, key) {
    this.id = id;
    this.key = key;
  }

  @computed
  get name() {
    const names = {
      house: "House Ad",
      adsense: "Google AdSense",
      dfp: "Google DFP",
      amazon: "Amazon",
      carbon: "Carbon",
      adbutler: "AdButler",
    };
    return names[this.key] || this.key;
  }
}

File: plugins/discourse-adplugin/lib/ad_plugin/ad_type.rb
# frozen_string_literal: true

module AdPlugin
  class AdType
    class << self
      def types
        @types ||= Enum.new(:house, :adsense, :dfp, :amazon, :carbon, :adbutler, start: 0)
      end

      def [](type)
        types[type]
      end

      def valid?(type)
        types.values.include?(type)
      end

      def enum_hash
        @enum_hash ||= types.to_h.freeze
      end
    end
  end
end
